src/main/java/uk/gov/hmcts/befta/util/MapVerifier.java

```diff
@@ -7,10 +7,27 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
-public class MapVerifier {
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANYTHING_IF_EXISTS;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANYTHING_PRESENT;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_DATE_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_DATE_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_FLOATING_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_FLOATING_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_INTEGER_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_INTEGER_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_NUMBER_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_NUMBER_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_OBJECT_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_OBJECT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_STRING_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_STRING_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_TIMESTAMP_NOT_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.ANY_TIMESTAMP_NULLABLE;
+import static uk.gov.hmcts.befta.util.MapPlaceholders.NOT_NULL;
 
-    private static final String ANY = "[[ANY]]";
-    private static final String NOT_NULL = "[[NOT_NULL]]";
+public class MapVerifier {
```

Comment:
In this class, we need to also check the types of the actual values against the types mandated by expected value place holders. For example, if the expected value is [[ANY_TIMESTAMP_NOT_NULLABLE]], we have to make sure there is a value in the actual field, and it is indeed in timestamp format.

Reply:
Changed implementation